packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -167,7 +198,13 @@ export async function _receiveCloudEventsDeserialize(
         id: p.event["id"],
         source: p.event["source"],
         data: p.event["data"],
-        dataBase64: Buffer.from(p.event["data_base64"] ?? ""),
+        dataBase64:
+          typeof p.event["data_base64"] === "string"
+            ? stringToUint8Array(
+                p.event["data_base64"] ?? new Uint8Array(),
+                "base64"
+              )
+            : p.event["data_base64"],
```

Comment:
This is because the rest layer p.event["data_base64"] is Uint8Array | string, and the modular layer is Uint8Array. So I use the typeof to decide if I need to convert it.